### PR TITLE
Add upload queue

### PR DESCRIPTION
**Pull request by leonholub**
## Purpose
Previously csv uploads where processed directly in each request, slowing down reaction time and causing issues with request timeouts because of long processing times. A background queue was introduced to reduce request time and process uploaded files in the background.

## Approach
On each upload the csv files get stored to the server and the basic silo/table structure gets created and the info is passed to a task in a celery queue which uses redis as a message broker to asynchronously process tasks.

### CODE REVIEW

**Comment by jefmoura** [comment]
@leonholub As we talked, it's good to implement now a model for the celery task, otherwise, it'll be hard to improve in the future with migrations and so on.
- Django Doc for [GenericForeignKey](https://docs.djangoproject.com/en/1.11/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey)

**Comment by leonholub** [comment]
@jefmoura agreed, i will add this and push changes later today.

**Comment by leonholub** [comment]
@Menda @jefmoura please review. I'm not sure about the usage of CeleryTasks in the view `siloDetail` it seems unecessary complicated but i couldn't find a better way to get the active tasks of the associated Reads with the `GenericForeignKey`.

File: silo/models.py [hunk 1]
```diff
@@ -250,6 +275,7 @@ class ReadTypeAdmin(admin.ModelAdmin):
     list_display = ('read_type','description','create_date','edit_date')
     display = 'Read Type'
 
+from django.contrib.contenttypes.fields import GenericRelation
```
Comment (Menda) [inline silo/models.py]: This would have gone better at the top of the file.

Comment (leonholub) [reply silo/models.py]: Oh yeah, right. I forgot to move that.

File: silo/views.py [hunk 2]
```diff
@@ -676,7 +679,7 @@ def showRead(request, id):
     """
     Show a read data source and allow user to edit it
     """
-    excluded_fields = ['gsheet_id', 'resource_id', 'token', 'create_date', 'edit_date', 'token', 'autopush_expiration', 'autopull_expiration']
+    excluded_fields = ['gsheet_id', 'resource_id', 'token', 'create_date', 'edit_date', 'token', 'autopush_expiration', 'autopull_expiration', 'task_id', 'task_status']
```
Comment (jefmoura) [inline silo/views.py]: The `Read` model doesn't have the fields `task_id` and `task_status` anymore. Remove them from here.

File: silo/models.py (hunk 1 above)
Comment (jefmoura) [reply silo/models.py]: Yes. Imports should be placed at the top.

**Comment by jefmoura** [comment]
I agree with @Menda, as you changed the `uploadFile()` and `siloDetail()`, we need to make sure via tests that they're working properly.

**Comment by leonholub** [comment]
I agree, we should test this thoroughly. I will add more tests.

**Pull request merged**